tests/unit/metrics/StatsClient.js

```diff
@@ -27,52 +28,110 @@ describe('StatsClient class', () => {
+                statsClient.getStats(fakeLogger, id, (err, res) => {
+                    assert.ifError(err);
+
+                    assert.equal(res.requests, 9);
+                    next();
+                });
+            },
+            next => {
+                statsClient.reportNewRequest(id);
+                statsClient.getStats(fakeLogger, id, (err, res) => {
+                    assert.ifError(err);
+
+                    assert.equal(res.requests, 10);
+                    next();
+                });
+            },
+            next => {
+                statsClient.reportNewRequest(id, noop);
+                statsClient.getStats(fakeLogger, id, (err, res) => {
+                    assert.ifError(err);
+
+                    assert.equal(res.requests, 11);
+                    next();
+                });
+            },
+        ], done);
     });
 
-    it('should correctly record a 500 on the server', () => {
+    it('should correctly record a 500 on the server', done => {
         statsClient.report500(id, (err, res) => {
             assert.ifError(err);
             assert(Array.isArray(res));
             assert.equal(res.length, 2);
 
             const expected = [[null, 1], [null, 1]];
             assert.deepEqual(res, expected);
+            done();
         });
     });
 
-    it('should respond back with total requests', () => {
-        statsClient.reportNewRequest(id, err => {
-            assert.ifError(err);
-        });
-        statsClient.report500(id, err => {
-            assert.ifError(err);
-        });
-        statsClient.getStats(fakeLogger, id, (err, res) => {
-            assert.ifError(err);
-            assert.equal(typeof res, 'object');
-            assert.equal(Object.keys(res).length, 3);
-            assert.equal(res.sampleDuration, STATS_EXPIRY);
+    it('should respond back with total requests', done => {
+        async.series([
+            next => {
+                statsClient.reportNewRequest(id, err => {
+                    assert.ifError(err);
+                    next();
+                });
+            },
+            next => {
+                statsClient.report500(id, err => {
+                    assert.ifError(err);
+                    next();
+                });
+            },
+            next => {
+                statsClient.getStats(fakeLogger, id, (err, res) => {
+                    assert.ifError(err);
+                    assert.equal(typeof res, 'object');
+                    assert.equal(Object.keys(res).length, 3);
+                    assert.equal(res.sampleDuration, STATS_EXPIRY);
 
-            const expected = { 'requests': 1, '500s': 1, 'sampleDuration': 30 };
-            assert.deepEqual(res, expected);
-        });
+                    const expected = {
+                        'requests': 1,
+                        '500s': 1,
+                        'sampleDuration': 30,
+                    };
+                    assert.deepEqual(res, expected);
```

Comment:
I'm not sure of the exact difference but we're mostly using `deepStrictEqual` for comparing objects deeply, maybe stick to this function to avoid surprises (it also applies to a few cases above).